skbio/stats/composition.py

```diff
@@ -590,23 +644,25 @@ def ilr(mat, basis=None, check=True):
     where rows represent basis vectors, and the columns represent proportions.
 
     """
-    mat = closure(mat)
+    xp = aac.array_namespace(mat)
     if basis is None:
         d = mat.shape[-1]
         basis = _gram_schmidt_basis(d)  # dimension (d-1) x d
-    else:
-        if len(basis.shape) != 2:
-            raise ValueError(
-                "Basis needs to be a 2D matrix, "
-                "not a %dD matrix." % (len(basis.shape))
-            )
-        if check:
-            _check_orthogonality(basis)
+    
+    assert isinstance(basis, xp.ndarray)
+    
+    if len(basis.shape) != 2:
+        raise ValueError(
+            "Basis needs to be a 2D matrix, "
+            "not a %dD matrix." % (len(basis.shape))
+        )
+    if check:
+        _check_orthogonality(basis)
 
-    return clr(mat) @ basis.T
+    return mat @ basis.T
 
 
-def ilr_inv(mat, basis=None, check=True):
+def ilr_inv(mat:Array, basis:Array|None=None, check:bool=True)-> Array:
```

Comment:
let's rename this to validate_args to be consistent with other packages (i.e. PyTorch). Every function will need to have this

Reply:
In other functions of scikit-bio this parameter is often called `validate`. [Example 1](https://scikit.bio/docs/latest/generated/skbio.diversity.beta_diversity.html) [Example 2](https://scikit.bio/docs/latest/generated/skbio.stats.distance.DissimilarityMatrix.html). I explored around and found that there isn't a consensus outside deep learning community. For example, SciPy's [curve_fit](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html) uses `check_finite`. scikit-learn doesn't have those flags but have standalone functions like [check_array](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_array.html), [check_X_y](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html), etc.